src/nisarqa/raster_classes.py

```diff
@@ -462,19 +462,25 @@ class GeoRaster(SARRaster):
         This corresponds to the lower side of the bottom pixels.
     epsg : int
         The EPSG code of the input raster.
+    x_coordinates : numpy.ndarray
+        1D vector of the coordinate values in the X direction for the
+        input array, starting from the left side of the left-most pixel
+        to the left side of the right-most pixel.
+    y_coordinates : numpy.ndarray
+        1D vector of the coordinate values in the Y direction for the
+        input array, starting from the upper edge of the top-most pixel
+        to the upper edge of the bottom-most pixel.
     """
 
     # Attributes of the input array
     x_spacing: float
-    x_start: float
-    x_stop: float
-
     y_spacing: float
-    y_start: float
-    y_stop: float
```

Comment:
`x_start` and friends are still attributes of the dataclass, but we're not going to pass them to the `__init__` method anymore. Let's use `dataclasses.field` to tell Python that these aren't going to be passed as arguments to the `__init__` method that it synthesizes for you, as demonstrated below:

```python
from dataclasses import field

@dataclass
class GeoRaster(SARRaster):

    x_start: float = field(init=False)
    x_stop: float = field(init=False)

    y_start: float = field(init=False)
    y_stop: float = field(init=False)
```